grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -183,9 +183,7 @@ class ProductController {
 
         if (!productInstance.hasErrors() && productInstance.save(flush: true)) {
             log.info("saved product " + productInstance.errors)
-            def warehouseInstance = Location.get(session.warehouse.id)
-            def inventoryInstance = warehouseInstance?.inventory
-            flash.message = "${warehouse.message(code: 'default.created.message', args: [warehouse.message(code: 'product.label', default: 'Product'), format.product(product: productInstance)])}"
+            flash.message = "${warehouse.message(code: 'default.created.message', args: [warehouse.message(code: 'product.label', default: 'Product').decodeHTML(), format.product(product: productInstance).decodeHTML()])}"
```

Comment:
We need exactly one `decodeHTML()` here for the same reason as in `CategoryController`, q.v.